### PR TITLE
Release 1.14: Version numbering and upgrade related changes

### PR DESCRIPTION
**Pull request by orhankislal**
Updates the version number to 1.14 for the release candidate.
Updates the changelists and other related files for upgrade.
Note that upgrade is not supported from versions prior to 1.11.

Co-authored-by: Nikhil Kak <nkak@pivotal.io>

### CODE REVIEW

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/madlib-pr-build/458/

File: src/madpack/changelist_1.13_1.14.yaml [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+# This file contains all changes that were introduced in a new version of
+# MADlib. This changelist is used by the upgrade script to detect what objects
+# should be upgraded (while retaining all other objects from the previous version)
+
+# New modules (actually .sql_in files) added in upgrade version
+# For these files the sql_in code is retained as is with the functions in the
+# file installed on the upgrade version. All other files (that don't have
+# updates), are cleaned up to remove object replacements
+new module:
+    # ----------------- Changes from 1.13 to 1.14 --------
```
Comment (iyerr3) [inline src/madpack/changelist_1.13_1.14.yaml]: Not sure if this is only for "module" or for any sql_in file - balance datasets is a new file added in this version.

File: src/madpack/upgrade_util.py [hunk 2]
```diff
@@ -82,16 +82,21 @@ def _get_function_info(self, oid):
...
+                        ELSE bar
                     END AS argname,
                     CASE array_upper(proargtypes,1) WHEN -1 THEN 1
-                        ELSE generate_series(0, array_upper(proargtypes, 1))
+                        ELSE zee
                     END AS i
                 FROM
-                    pg_proc AS p
+                    (SELECT *, oid,
+                        unnest(proargtypes)::regtype AS foo,
```
Comment (iyerr3) [inline src/madpack/upgrade_util.py]: why the non-meaningful names? And while we're here, could we please add some comments on what's happening in the query?

Comment (orhankislal) [reply src/madpack/upgrade_util.py]: We changed the query because PG 10 does not allow set returning functions in the `CASE` clause any more.

**Comment by orhankislal** [comment]
Thanks for the comments Rahul.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/madlib-pr-build/460/

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/madlib-pr-build/461/

**Pull request closed without merging**